DistroLauncher/upgrade_policy.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "stdafx.h"
+
+bool starts_with(std::wstring_view tested, std::wstring_view start)
+{
+    if (tested.size() < start.size()) {
+        return false;
+    }
+    auto mismatch = std::mismatch(start.cbegin(), start.cend(), tested.cbegin());
+    return mismatch.first == start.cend();
+}
+
+bool ends_with(std::wstring_view tested, std::wstring_view end)
+{
+    if (tested.size() < end.size()) {
+        return false;
+    }
+    auto mismatch = std::mismatch(end.crbegin(), end.crend(), tested.crbegin());
+    return mismatch.first == end.crend();
+}
+
+std::wstring GetUpgradePolicy()
+{
+    std::wstring_view name = DistributionInfo::Name;
+    if (name == L"Ubuntu") {
+        return L"lts";
+    }
+    if (starts_with(name, L"Ubuntu") && ends_with(name, L"LTS")) {
+        return L"never";
+    }
+    return L"normal";
+}
+
+void SetDefaultUpgradePolicyImpl()
+{
+    namespace fs = std::filesystem;
+
+    const fs::path log{L"/var/log/upgrade-policy-changed.log"};
+    const fs::path policyfile{L"/etc/update-manager/release-upgrades"};
+
+    if (fs::exists(log)) {
```

Comment:
Windows will never find that file without prefixing its path with the WSL prefix.

Reply:
Oops. Fixed.